File: codyze_v3/codyze-specification-languages/mark/src/main/java/de/fraunhofer/aisec/codyze/mark/analysis/TypestateMode.java

package de.fraunhofer.aisec.codyze.mark.analysis;

public enum TypestateMode {

	/**
	 * Non-deterministic finite automaton. Intraprocedural, not alias-aware.
	 */
	DFA,

	/**
	 * Weighted Pushdown System. Interprocedural, alias-aware, context-aware.
	 */
	WPDS

}

File: codyze_v3/codyze-core/src/todo/config/TypestateMode.kt
package de.fraunhofer.aisec.codyze_core.config

enum class TypestateMode {
    // Non-deterministic finite automaton. Intraprocedural, not alias-aware.
    DFA,

    // Weighted Pushdown System. Interprocedural, alias-aware, context-aware.
    WPDS
}
